infullmvp-kotlin-koin/src/main/java/com/infullmobile/android/infullmvp/InFullMvpFragment.kt

```diff
@@ -0,0 +1,70 @@
+package com.infullmobile.android.infullmvp
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.support.annotation.CallSuper
+import android.support.v4.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import org.koin.android.ext.android.releaseContext
+import org.koin.android.scope.ext.android.bindScope
+import org.koin.android.scope.ext.android.getOrCreateScope
+
+abstract class InFullMvpFragment<VIEW : MvpView<PRESENTER>, out PRESENTER : MvpPresenter<VIEW>>
+    : Fragment(), MvpView<PRESENTER> {
+
+    abstract val scopeName: String
+    abstract val layoutId: Int
+    override val activity: Activity get() = getActivity()!!
```

Comment:
Exposing context as a value was a helping hand in getting the resources out most of the time. Exposing an activity may lead to a lot of lazy code (cast to specific implementation, issue some method).

Please use Context.